Looks like two tests are failing:

```
FAILED tests/unit/test_dbapi_cursor.py::TestCursor::test_fetchall_w_bqstorage_client_fetch_error_no_fallback
FAILED tests/unit/test_dbapi_cursor.py::TestCursor::test_fetchall_w_bqstorage_client_no_arrow_compression
```

I think we need to remove this logic:

https://github.com/googleapis/python-bigquery/blob/77d71736fcc006d3ab8f8ba17955ad5f06e21876/google/cloud/bigquery/dbapi/cursor.py#L380-L384

Page size of 1 is going to be super slow, so I don't think we should be setting that if people explicitly request "None" as the page size.

--- a/google/cloud/bigquery/table.py
+++ b/google/cloud/bigquery/table.py
@@ -1873,6 +1873,11 @@ class RowIterator(HTTPIterator):
         """total bytes processed from job statistics, if present."""
         return self._total_bytes_processed
 
+    @property
+    def page_size(self) -> Optional[int]:
+        """The maximum number of rows in each page of results from this request, if present."""
+        return self._page_size
+
     def _is_almost_completely_cached(self):
         """Check if all results are completely cached.
 
@@ -1924,7 +1929,7 @@ class RowIterator(HTTPIterator):
         if self._is_almost_completely_cached():
             return False
 
-        if self.max_results is not None:
+        if self.max_results is not None or self.page_size is not None:
             return False
 
         try:
@@ -1994,7 +1999,9 @@ class RowIterator(HTTPIterator):
             bqstorage_client:
                 The BigQuery Storage client intended to use for downloading result rows.
         """
-        if bqstorage_client is not None and self.max_results is not None:
+        if bqstorage_client is not None and (
+            self.max_results is not None or self.page_size is not None
+        ):
             warnings.warn(
                 "Cannot use bqstorage_client if max_results is set, "
                 "reverting to fetching data with the REST endpoint.",

--- a/tests/unit/test_dbapi_cursor.py
+++ b/tests/unit/test_dbapi_cursor.py
@@ -161,6 +161,7 @@ class TestCursor(unittest.TestCase):
             mock_rows,
         )
         mock_rows.max_results = None
+        mock_rows.page_size = None
         type(mock_rows).job_id = mock.PropertyMock(return_value="test-job-id")
         type(mock_rows).location = mock.PropertyMock(return_value="test-location")
         type(mock_rows).num_dml_affected_rows = mock.PropertyMock(

--- a/tests/unit/test_table.py
+++ b/tests/unit/test_table.py
@@ -2693,6 +2693,13 @@ class TestRowIterator(unittest.TestCase):
         )
         self.assertFalse(result)
 
+    def test__should_use_bqstorage_returns_false_if_page_size_set(self):
+        iterator = self._make_one(page_size=10, first_page_response=None)  # not cached
+        result = iterator._should_use_bqstorage(
+            bqstorage_client=None, create_bqstorage_client=True
+        )
+        self.assertFalse(result)
+
     def test__should_use_bqstorage_returns_false_w_warning_if_missing_dependency(self):
         iterator = self._make_one(first_page_response=None)  # not cached
 

